---
title: "HiSeq Biogeography - Script 3, Main Figures" 
author: "Diana Proctor"
date: "December 13, 2017"
output: html_document


---
Supplementary Data File 3 - 

This is the THIRD script used to generate the main figures for the the manuscript by Proctor et al. titled "A spatial gradient of bacterial diversity in the human oral cavity shaped by salivary flow" 


<WORK> This script and associated data are provided via (c) by Diana M Proctor, Julia A. Fukuyama, Susan P. Holmes, David A. Relman. These data and the associated script are licensed under the Creative Commons Attribution-ShareAlike 4.0 International License (CC-BY-CA).

Given attribution, you are free to:
1) Share, copy and redistribute the material in any medium or format
2) Adapt, remix, transform, and build upon the material for any purpose, even commercially.

To see the full license associated with attribution of this work, see the CC-By-CA license,  see <http://creativecommons.org/licenses/by-sa/4.0/>.


#### this script is used to generate the main figures for the manuscript
Run the decontamination script (Supplementary Data File 1) first. Then run this one. Most of the supplemental figures are in an accompanying files (Supplementary Data Files 3-4). The local name of this file on my desktop is: Proctor_MainFigures_v28.0.Rmd.


### Import the data (from the decontamination script) and create a supragingival dataset

```{r, admin, fig.height=8.5, fig.width=11, warning=FALSE, echo=FALSE}
library("phyloseq");library("ggplot2");library(gridExtra);library("stringr");library("reshape2");library("genefilter"); library(knitr);library(DESeq2)
setwd("~/Desktop/Proctor_NatureComm/")
decontam.phy = readRDS("decontam.phy.rds")

decontam.phy = subset_samples(decontam.phy, Tooth_Class !="EC")
decontam.phy = subset_samples(decontam.phy, Tooth_Class !="NotApplicable")

#Create a supragingival tooth subset
supra <- subset_samples(decontam.phy, Habitat_Class=="Supra")
        supra <- prune_taxa(taxa_sums(supra) > 0, supra) 
        supra <- prune_samples(sample_sums(supra) > 0, supra)
        supra 

#this was output and then taxa were blast individuallly
#assignments were made if a single blast hit had the best escore for 99% identity 
#highest taxonomic ranks were assigned as genus or species and numbers were arbitrarily designated
#write.csv(data.frame(tax_table(supra)), file="~/Desktop/supra_tax.csv")
tax = read.csv("~/Desktop/Proctor_NatureComm/supra_phyloseq_tax_blast_20170516.csv")
rownames(tax) = tax$X
tax = tax_table(as.matrix(tax))


#replace the taxa table with the one with highest ranks
tax_table(supra) = tax
```

### 0a. Description of data: Load in the discovery dataset and get summary stats for the dataset
```{r}
load("~/Desktop/Proctor_NatureComm/mouth.RData")
discovery = mouthPhy

#how many samples and taxa
discovery

#what is the sequencing depth?
summary(sample_sums(discovery))

#how many subjects are in this dataset?
length(levels(as.factor(sample_data(discovery)$Subject)))
```


### 0b. Description of data: Load in the validation dataset and get summary stats for the dataset
```{r}
#how many samples are in the validation dataset?
nsamples(supra)

#how many subjects are in the validation dataset?
length(levels(sample_data(supra)$Subject))

#how many healthy aim 1 subjects are in the validation dataset?
aim1 = subset_samples(supra, Aim=="SA1" | Aim=="Pilot")
length(levels(sample_data(aim1)$Subject))

#how many healthy aim 3 subjects are in the validation dataset?
aim3 = subset_samples(supra, Aim=="SA3")
length(levels(sample_data(aim3)$Subject))

#what is the median sequencing depth of the validation dataset?
summary(sample_sums(supra))

```

### 0.c. Description of data: Load in the mucosal biogeography dataset and get the sequencing statistics
```{r, fig.width=12, fig.height=9}
library(vegan)
load("~/Desktop/Proctor_NatureComm/Biogeo1_phys.Rdata")
#fix the sample names in the Biogeo object
names <- sample_names(Biogeo1_phys)
names <- str_replace_all(names, ("aaa"), ".")
names <- str_replace_all(names, ("zzz"), ".")
sample_names(Biogeo1_phys) <- names
#use version 5 of the mapping file, includes coordinates
map <- import_qiime(mapfilename="~/Desktop/Proctor_NatureComm/biogeo_mapping_file_corrected_v5.0.txt")

sample_data(Biogeo1_phys) <- map

sites = c("AG", "BM", "KG", "Supra")
#subset on the sites analyzed in this study
Biogeo2_phys <- subset_samples(Biogeo1_phys, Site.New %in% sites)
sample_data(Biogeo2_phys)$TissueClass = ifelse(sample_data(Biogeo2_phys)$Site.New =="Supra", "Non-shedding            ", "Shedding               ")

#note that alveolar mucosa and keratinized gingiva sites are mislabeled; switch them
### Fix the other sample site labels
sample_data(Biogeo2_phys)$Site.New = str_replace_all(sample_data(Biogeo2_phys)$Site.New, "(AG)", "Keratinized Gingiva   ") 
sample_data(Biogeo2_phys)$Site.New = str_replace_all(sample_data(Biogeo2_phys)$Site.New, "(BM)", "Buccal Mucosa   ")
sample_data(Biogeo2_phys)$Site.New = str_replace_all(sample_data(Biogeo2_phys)$Site.New, "(KG)", "Alveolar Mucosa   ")
sample_data(Biogeo2_phys)$Site.New = str_replace_all(sample_data(Biogeo2_phys)$Site.New, "(Supra)", "Supragingival Tooth   ")

#how many samples are in the mucosal biogeo dataset?
nsamples(Biogeo2_phys)


#library(dada2)
#sequence table from rdp classifier on rdp database
#tax = tax_table(Biogeo1_phys)
#set.seed(100) # Initialize random number generator for reproducibility
#taxa <- assignTaxonomy(rownames(tax), "~/Dropbox/rdp_train_set_14.fa.gz", minBoot=80)
#taxaSpecies <- addSpecies(taxa, "~/Dropbox/rdp_species_assignment_14.fa.gz", verbose=TRUE)
#tax_table(Biogeo1_phys) <- taxaSpecies


#blast some of the sequences and create a new variable column for the highest taxonomic rank
#write.csv(data.frame(tax_table(Biogeo1_phys), taxa_sums(Biogeo1_phys)), file="biogeo_species_v1.0.csv")

#read in the new taxonomic table
tax2 = read.csv("~/Desktop/Proctor_NatureComm/biogeo_species_v3.0.csv", header=TRUE, row.names=1)
tax3 = tax_table(tax2)
taxa_names(tax3) = rownames(tax2)
colnames(tax3) = colnames(tax2)
tax_table(Biogeo2_phys) = tax3

#relabel with the highest rank designation
taxa_names(Biogeo2_phys) = tax_table(Biogeo2_phys)[,8]

#filter to retain taxa present in 15% of samples
Biogeo2_phys15 = filter_taxa(Biogeo2_phys, function(x) sum(x > 10) > (0.15 * length(x)),  TRUE)
Biogeo2_phys15 = prune_taxa(taxa_sums(Biogeo2_phys15) > 0, Biogeo2_phys15)
Biogeo2_phys15 = prune_samples(sample_sums(Biogeo2_phys15) > 0, Biogeo2_phys15)
Biogeo2_phys15 

### Set up several otu tables with different transformations
#raw data

#vst-transformed data
### Stabilize the variance
#create a deseq object
Biogeo2_VST <- Biogeo2_phys15
otu_table(Biogeo2_VST) <- otu_table(Biogeo2_VST) + 1
Biogeo2.ds = phyloseq_to_deseq2(Biogeo2_VST, ~Subject + Site.New)

#variance stabilizing transformation
Biogeo2_phys15_VST <- varianceStabilizingTransformation(Biogeo2.ds , blind=TRUE, fitType="local") 
biogeo_counts_VST <- otu_table(as.matrix(assay(Biogeo2_phys15_VST)), taxa_are_rows=TRUE)
otu_table(Biogeo2_VST) <- biogeo_counts_VST 
otus.VST = data.frame(t(otu_table(Biogeo2_VST)))

#how many samples are in the mucosal biogeo dataset?
nsamples(Biogeo2_phys)

#how many supra samples and mucosal samples are there
sb = data.frame(sample_data(Biogeo2_phys))
table(sb$Site.New)

#how many mucosal samples are there
sb = subset(sb, Site.New != "Supragingival Tooth   ")
sum(table(sb$Site.New))

#how many subjects / samples per subject are in the mucosal biogeo dataset?
table(sb$Subject)

#what is the median sequencing depth of the mucosal biogeo dataset?
summary(sample_sums(Biogeo2_phys))
```

#### 0.d how many samples are in the combined data?
```{r}
#how many taxa are in the decontaminated validation data
supra
nsamples(discovery)+nsamples(Biogeo2_phys)+nsamples(supra)
```

### 0.e. Description of data: Phyla level composition survey
- These results are described in the main text 
- Supplementary Table 1: Top 5 Phyla account for most of the reads

```{r}
#What is the relative abundance of each Phylum?
library(doBy)
supra.phy <- tax_glom(supra, taxrank="Phylum")
tax.count <- data.frame(tax_table(supra.phy)[,2:9], taxa_sums(supra.phy))
colnames(tax.count) <- c("Kingdom", "Phylum","Class","Order","Family","Genus", "Species", "Highest",  "Abundance")
Phylum_df <- summaryBy(Abundance~Phylum, data=tax.count, FUN=sum)
Phylum_df $Percent <- round(Phylum_df $Abundance.sum/sum(Phylum_df $Abundance)*100, 4)
library(plyr)
Phylum_df <- arrange(Phylum_df, desc(Percent))
Phylum_df
write.csv(Phylum_df, file="~/Desktop/TableS1.csv")
#how much do the top 5 phyla contribute to total abundance?
top5 <- Phylum_df[1:5, ]
round(sum(top5$Percent),2)
#What are the top 5 Phyla?
top5
```

### 0.f. Description of data: genus-level summary of data
- This is described in the main text
- Supplementary Table 2: Top 10 Genera account for most of the reads

```{r}
#What is the number of Genera found?
length(get_taxa_unique(supra, taxonomic.rank="Genus"))

#what are the abundance levels of each genus?
supra.genus <- tax_glom(supra, taxrank="Genus")
tax.count <- data.frame(tax_table(supra.genus)[,2:9], taxa_sums(supra.genus))
colnames(tax.count) <- c("Kingdom", "Phylum","Class","Order","Family","Genus", "Species", "Highest",  "Abundance")
Genus_df<- summaryBy(Abundance~Genus, data=tax.count, FUN=sum)
Genus_df$Percent <- round(Genus_df$Abundance.sum/sum(Genus_df$Abundance)*100, 4)
library(plyr)
Genus_df <- arrange(Genus_df, desc(Percent))
Genus_df <- Genus_df[order(Genus_df$Percent, decreasing=TRUE), ] 

#how much do the top 10 genera contribute to total abundance?
top10 <- Genus_df[1:10, ]
round(sum(top10$Percent),3)

#what about the top 5
top5 <- Genus_df[1:5, ]
round(sum(top5$Percent),3)

###How diverse are the top 10 genera? i.e., how many species are there per genus?
top10 <- as.vector(Genus_df$Genus[1:10])
Diversity.list <- vector("list", 10)
names(Diversity.list) <- top10

for(i in 1:length(top10)){
       physub = subset_taxa(supra, Genus==top10[i])
       physub = prune_taxa(taxa_sums(physub) > 0, physub)
       Diversity.list[[i]] <- physub
}

#compute the number of taxa in each element of the list
Ntaxa <- data.frame(unlist(lapply(Diversity.list, ntaxa)))

colnames(Ntaxa) <- "N.Species"
#Make a table with percent abundance and number of taxa
genus.tab <- data.frame(Genus_df[1:10,], Ntaxa)
library(knitr)
kable(genus.tab, format="markdown")
write.csv(genus.tab, file="~/Desktop/TableS2.csv")

```


### 1a. Perform PCoA on Bray Curtis and use an adonis to partitiion variation in to time, tooth class, tooth aspect, and subject
- This dataset consists of just molars and incisors 
- We also subset on taxa present at 25% prevalence since this is about 2x the number of samples per mouth
- Other prevalence thresholds seem to yield comparable results
- the scree plot indicates we should evaluate the first four axes

```{r, Figure 1, fig.width=12, fig.height=8, warning=FALSE}
library(RColorBrewer)
### Create an index tooth biogeography subset
index <- subset_samples(supra, Protocol=="Home")
sites <- c("Molar", "Incisor_Central")
index_subsites <- subset_samples(index, Tooth_Class %in% sites)
index_subsites
        
#how many samples are in average 2 subjects? Looks like about 25% of samples
mean(table(sample_data(index_subsites)$Subject))

#filter the taxa to include those present in 25% of samples
filtergroup = filterfun(kOverA(k=850, A=1)) #k = number of samples; A = abundance
        index25 = filter_taxa(index_subsites, filtergroup, prune=TRUE) 
        index25 = prune_samples(sample_sums(index25) > 0, index25) 

        #note the following command drops 7 samples (depth between 1-35)
        index25 = prune_samples(sample_sums(index25) > 800, index25) 
        index25
        
#fix the labeling of the subject ids 
#relabel for prettier plotting
sample_data(index25)$Tooth_Class <- str_replace_all(sample_data(index25)$Tooth_Class, "(Incisor_Central)", "Incisor")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(1-101)", "Subject 1")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(1-102)", "Subject 2")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(1-104)", "Subject 3")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(1-105)", "Subject 4")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(1-106)", "Subject 5")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(1-107)", "Subject 6")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(P1-8)", "Subject 7")
sample_data(index25)$Subject <- str_replace_all(sample_data(index25)$Subject, "(P1-9)", "Subject 8")

#force the labeling of the subjects
ordering= c("Subject 1", "Subject 2",  "Subject 3" , "Subject 4" , "Subject 5" ,"Subject 6", "Subject 7" ,"Subject 8")
sample_data(index25)$Subject  <- factor(sample_data(index25)$Subject, levels = ordering)


#hellinger transform the data
otus = data.frame(otu_table(index25))     
library(vegan)
otus.h = decostand(otus, "hellinger")  

#Do PCOA
#nmds on bray curtis distance of hellinger transformation
h.euc = vegdist(otus.h, "bray")
h.nmds = cmdscale(h.euc, eig=TRUE, k=10)
h.map = data.frame(sample_data(index25), scores(h.nmds))
h.map$method = "Hellinger"
h.eig = 100*(h.nmds$eig/sum(h.nmds$eig))
head(h.eig)

#make a screeplot
evals <- h.nmds$eig
foo = data.frame(h.eig, as.numeric(1:length(h.eig)))
colnames(foo) = c("percent.variation", "rank")
p=ggplot(foo[1:10,], aes(rank, percent.variation)) + geom_point() + theme_bw()
p
```



#### 1c. Figure 1: Samples segregate by tooth class along the first axis
```{r}
#look at all the data as a function of tooth class
p1=plot_ordination(index25, h.nmds, type="samples", color="Tooth_Class") + geom_point(size=2, alpha=0.1) + theme_bw() +theme(legend.position = "none") + theme(text = element_text(size=18), axis.text.x = element_text(angle=0, vjust=1))+coord_fixed(sqrt(evals[2] / evals[1]))
p.dat = p1$data

##### plot figure panels
#1a
fig1a=plot_ordination(index25, h.nmds, type="samples", color="Subject") + geom_point(alpha=0.4) + theme_bw() + theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) + scale_colour_brewer(palette="Dark2")+coord_fixed(sqrt(evals[2] / evals[1]))  + ggtitle("a)") + xlab("Axis 1 (23.48%)") + ylab("Axis 2 (10.99%)")


#1b
fig1b=plot_ordination(index25, h.nmds, type="samples", color="Tooth_Class")   + geom_point(alpha=0.4)  + theme_bw() + theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))  +coord_fixed(sqrt(evals[2] / evals[1])) + guides(color=guide_legend(title="Tooth Class")) + ggtitle("b)")  + xlab("Axis 1 (23.48%)") + ylab("Axis 2 (10.99%)")

#1c
fig1c=plot_ordination(index25, h.nmds, type="samples", color="Tooth_Aspect")   + geom_point(alpha=0.4) + theme_bw() + theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) + scale_colour_brewer(palette="Set1") +coord_fixed(sqrt(evals[2] / evals[1]))+ guides(color=guide_legend(title="Tooth Aspect")) + ggtitle("c)")  + xlab("Axis 1 (23.48%)") + ylab("Axis 2 (10.99%)")

#1d
fig1d <- ggplot(p.dat, aes(Tooth_Class, Dim1, fill=Tooth_Class)) + geom_boxplot(notch=TRUE) + facet_wrap(~Tooth_Aspect) + theme_bw() + theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) +  ylab("Axis 1 (23.48%)") + xlab("")+ guides(fill=guide_legend(title="Tooth Class")) + ggtitle("d)") 

grid.arrange(fig1a, fig1b, fig1c, fig1d, ncol=2)

#save figure 1
ggsave(grid.arrange(fig1a, fig1b, fig1c, fig1d, ncol=2), file="~/Dropbox/Figure1.eps",  width = 10, height = 8, units ="in",dpi = 300, device="eps")


#how many comparisons are there in the boxplots
#buccal
b = subset(p.dat, Tooth_Aspect=="Buccal")
table(b$Tooth_Class)
#lingual
l = subset(p.dat, Tooth_Aspect=="Lingual")
table(l$Tooth_Class)
```

#### 1d. Quantify the separation of samples along the axes based on tooth class and tooth aspect
```{r}
p1.dat = fig1d$data

#how many incisors have negative scores
test1 = subset(p1.dat, Tooth_Class=="Incisor" & Dim1 > 0)
testb = subset(p1.dat, Tooth_Class=="Incisor")
nrow(test1)/nrow(testb)

#of incisors with + scores, many incisors have scores > than 0.25?
test2 = subset(p1.dat, Tooth_Class=="Incisor" & Dim1 > 0.25)
nrow(test2)/nrow(test1)

#how many molar communities have positive scores 
test3 = subset(p1.dat, Tooth_Class=="Molar" & Dim1 > 0)
testb = subset(p1.dat, Tooth_Class=="Molar")
nrow(test3)/nrow(testb)

#how many molars have scores > than 0.25
test4 = subset(p1.dat, Tooth_Class=="Molar" & Dim1 > 0.25)
testb = subset(p1.dat, Tooth_Class=="Molar" ) 
nrow(test4)/nrow(test3)


## see how many lingual samples have scores less than 0.25 sicnce there seems to be an interaction with most lingual incisors falling below 0.25
test5 = subset(p1.dat, Tooth_Aspect=="Lingual" & Dim1 > 0)
testb = subset(p1.dat, Tooth_Aspect=="Lingual" ) 
nrow(test5)/nrow(testb)

```

#### 1e. Merge temporal replicates by subject so that we can test the nmds model using Adonis 

```{r, Fig 1 test, fig.width=12, fig.height=6, warning=FALSE}
#merge the temporal replicates within each subject to avoid temporal pseudoreplication        
subs <- levels(sample_data(index25)$Subject) 
holder <-vector('list',length(subs)) 


for(i in 1:length(subs)){ 
    subx = subset_samples(index25, Subject==subs[[i]])
    subh = merge_samples(subx, "Habitat")
    sample_data(subh)$Subject = subs[[i]]
    sample_data(subh)$Habitat = sample_names(subh)
    sample_names(subh) = paste0(sample_data(subh)$Subject, "_", sample_names(subh))
    holder[[i]] = subh
}

is = merge_phyloseq(holder[[1]], holder[[2]], holder[[3]], holder[[4]], holder[[5]], holder[[6]], holder[[7]])
is2 = prune_taxa(taxa_sums(is) > 0, is)
is2 = prune_samples(sample_sums(is2) > 0, is2)
is2

#set up a new mapping file to correct what was lost during the merge
        foo = sample_names(is2)
        foo2 <- str_replace_all(foo, "([B,L])", replacement =c(".Buccal", ".Lingual"))
        foo2 <- str_replace_all(foo2,  "_", ".")
        map <- colsplit(foo2, "([.])", c("Subject", "Junk", "Tooth_Number", "Tooth_Aspect"))
        map = data.frame(sample_data(is2)$Habitat, map)
        colnames(map)[1] = "Habitat"
         
        #merge in the mapping file information
        dot = read.csv("~/Desktop/Proctor_NatureComm/mytoothdot_coordinates_FullMouth.csv")
        colnames(dot)[1] = "Habitat"
        
        #merge maps
        library(plyr)
        map2 = join(map, dot, by="Habitat")
        
        #make the sample names of map2 the same as the names in bs2 then replace the mapping file
        names = paste0(map2$Subject, "_", map2$Habitat)
        rownames(map2) = names
        sample_data(is2) = map2
```

### 1f. Use adonis to see how much variation each factor explains after adjusting for temporal replication
- Note: we have time series here so to avoid violation of the rule of non-independence of samples we'll sum across these replicates
- Then we'll see how much the factors subject, tooth class and tooth aspect explain variation in Bray Curtis dissimilarity, performing a permutation analysis, stratifying within subjects
- One thing of interest is the interaction between tooth class and tooth aspect since the lingual incisors seem to separate more from the molar samples than the buccal incisor samples

```{r}
#library(phyloseq)
library(vegan)
otus = data.frame(otu_table(is2))
otus.h = decostand(otus, "hellinger")
otus.euc = vegdist(otus.h, "euclidean")
attach(sample_data(is2))
set.seed(12)
index.adonis <- adonis(otus.euc ~Subject*Tooth_Class*Tooth_Aspect, permutations=1000, strata = Subject) 
df <- format(data.frame(index.adonis$aov.tab), digits=2)
kable(df, format="markdown")
#write.csv(df, file="~/Dropbox/Figures_20170617/Revised/FinalFigures/TableS3.csv")
detach(sample_data(is2))
```


### 1g. Check to see if sequencing depth explains these differences between groups
```{r}
#do we see a difference in sequencing depth by tooth location, no we don't for most subjects 
df = data.frame(sample_sums(is2), sample_data(is2))
colnames(df)[1] = "abundance"

#does sequencing depth vary by tooth class on lingual surfaces?
ling= subset(df, Tooth_Aspect=="Lingual")
wilcox.test(abundance~Tooth_Class, data=ling, exact=TRUE, conf.int=TRUE)
#whatabout buccal surfaces
buc= subset(df, Tooth_Aspect=="Buccal")
wilcox.test(abundance~Tooth_Class, data=buc, exact=TRUE, conf.int=TRUE)

```


### 1f. Does alpha diversity differ between the buccal and lingual tooth aspects in human health?
- it seems to differ for the lingual samples
- we want to know whether this is also true for clinic samples 
- if it is then this suggests that the differences we are seeing may be a result of difference in biomass sampled
```{r, fig.width=12, fig.height=6, warning=FALSE}
mi= subset_samples(supra, Tooth_Class %in% c("Incisor_Central", "Molar"))
mi = prune_samples(sample_sums(mi) > 0, mi)


meths = c("Shannon", "Chao1", "Simpson")
sites = levels(as.factor(sample_data(mi)$Habitat))
holder <-vector('list',length(sites)) 

for(i in length(sites))
  {
  # Estimate diversity and make data frame
  erDF <- estimate_richness(mi, split = TRUE, measures = meths)
  df <- data.frame(erDF, sample_data(mi))
  holder[[i]] <- df
}

df <- data.frame(do.call("rbind", holder)) 


#test to see whether alpha diversity varies based on tooth class
wilcox.test(Chao1~Tooth_Class, data=df, exact=TRUE, conf.int=TRUE)
wilcox.test(Shannon~Tooth_Class, data=df, exact=TRUE, conf.int=TRUE)
wilcox.test(Simpson~Tooth_Class, data=df, exact=TRUE, conf.int=TRUE)
```


#### 1g. is this true for clinic samples only?
- subjects often report that it is difficult to collect lingual samples
- we want to know whether this is true of the clinic samples only
- we see that it isn't so the differences that we are seeing are likely not a result of differences in alpha diversity

```{r, fig.width=12, fig.height=6, warning=FALSE}
mi= subset_samples(mi, Protocol=="Clinic")
mi = prune_samples(sample_sums(mi) > 0, mi)
mi

meths = c("Shannon", "Chao1", "Simpson")
sites = levels(as.factor(sample_data(mi)$Habitat))
holder <-vector('list',length(sites)) 

for(i in length(sites))
  {
  # Estimate diversity and make data frame
  erDF <- estimate_richness(mi, split = TRUE, measures = meths)
  df <- data.frame(erDF, sample_data(mi))
  holder[[i]] <- df
}

df <- data.frame(do.call("rbind", holder)) 


#test to see whether there are two class differences
#test to see whether alpha diversity varies based on tooth class
wilcox.test(Chao1~Tooth_Class, data=df, exact=TRUE, conf.int=TRUE)
wilcox.test(Shannon~Tooth_Class, data=df, exact=TRUE, conf.int=TRUE)
wilcox.test(Simpson~Tooth_Class, data=df, exact=TRUE, conf.int=TRUE)


```



#### 2.1. Figure 2: Test to see if this variation is consistent with an ecological gradient across all tooth classes
- in this case, we will sum temporal replicates to deal with the problem of pseudoreplication

```{r, fig.width=6, fig.height=6, warning=FALSE} 
#### 1. Compute the third-order orthogonal polynomial terms
#create validation datset #2 for time x space analysis, all teeth, healthy subjects
keep <- c("1-101", "1-102", "1-103", "1-104", "1-105", "1-106", "1-107")
FullMouth <- subset_samples(supra, Subject %in% keep & Protocol=="Clinic")
FullMouth <- prune_taxa(taxa_sums(FullMouth) > 0, FullMouth)
FullMouth

#filter the data
filtergroup = filterfun(kOverA(k=200, A=1)) #k = number of samples; A = abundance
        f1 = filter_taxa(FullMouth, filtergroup, prune=TRUE) 
        f1 = prune_taxa(taxa_sums(f1) > 0, f1)
        f1 = prune_samples(sample_sums(f1) > 0, f1)
        f1

#fix the variable labeling for prettier plotting
sample_data(f1)$Tooth_Class <- str_replace_all(sample_data(f1)$Tooth_Class, "(Incisor_Central)", "Incisor")
sample_data(f1)$Tooth_Class <- str_replace_all(sample_data(f1)$Tooth_Class, "(Incisor_Lateral)", "Incisor")
sample_data(f1)$Tooth_Class <- str_replace_all(sample_data(f1)$Tooth_Class, "(Molar_Pre)", "Premolar")

#hellinger transform the data
library(vegan)
otus = data.frame(otu_table(f1))
otus.h = decostand(otus, "hellinger")
map = sample_data(f1)
map$x = as.numeric(as.character(map$x))
map$y = as.numeric(as.character(map$y))

#center the coordinates
xygrid = cbind(map$x, map$y)
xygrid.c <- scale(xygrid, scale=FALSE)

### Compute the third-order orthogonal polynomial function using centered geographic coordinates
poly.xy3 <- poly(xygrid.c, degree = 3, raw=FALSE) #, coefs=TRUE) 
colnames(poly.xy3) <- c("X", "X^2", "X^3", "Y", "XY", "X^2Y", "Y^2", "XY^2", "Y^3")
poly.xy3.df <- data.frame(poly.xy3, map$x, map$y)
```


#### 2.2. Perform trend surface analysis 
- use the 9 polynomial terms - note: this is comparable to what we've already done, but instead of using VST transformed data we use the hellinger transformed data as the reviewer suggested
- use the hellinger transformed data which doesn't weight rarer or under-sampled species
- we see that the pattern is the same, but there are more outliers with this transformation than with the other
- we see that the molars and incisors tend to separate out on the first main axis 
```{r}
library(ade4)
#perform the RDA on the hellinger transformed data; extract the coordinates 
rld.pca <- dudi.pca(otus.h , center=TRUE, scale=TRUE, scannf=FALSE, nf=10)
rld.xy3 <- pcaiv(rld.pca, poly.xy3, scannf = FALSE, nf = 6)
rld.xy3.df <- data.frame(rld.xy3$ls, map)
xy3.df <- data.frame(rld.xy3.df, poly.xy3)

# how much of the variance does the trend surface model explain?
rld.xy3.var <- sum(rld.xy3$eig)/sum(rld.pca$eig)*100
rld.xy3.var

#look at the eigenvalues
rld.xy3$eig

#look at the screeplot
screeplot(rld.xy3) #we should look at the first 3 axes


#what is the percent of ewxplained variance
Explainedvariance = rld.xy3$eig/sum(rld.xy3$eig)*100
Explainedvariance

#force x, y to numeric
xy3.df$x <- as.numeric(as.character(xy3.df$x))
xy3.df$y <- as.numeric(as.character(xy3.df$y))

### Force variables to order in ggplot
order=1:32
xy3.df$Tooth_Number <- factor(xy3.df$Tooth_Number, as.character(order))
xy3.df$tclass = xy3.df$Tooth_Class
xy3.df$tclass = str_replace_all(xy3.df$tclass, "Incisor_Central", "Incisor")
xy3.df$tclass = str_replace_all(xy3.df$tclass, "Incisor_Lateral", "Incisor")
```


#### 2.3 it looks like the tooth classes separate along axis 1 in some subjects; let's look at variation of tooth classes against the spatial coordinates
- we see that the molars and incisors tend to separate along axis 1
- we also see that this pattern holds true in the case where we are using hellinger transformed data; these findings are consistent with other transformations, including the variance stabilized transformation as well as the rank-abundance transformation
````{r}
#plot axis 1
cbPalette <- c("grey76", "Salmon", "#00BFC4", "grey76")
a1a = ggplot(xy3.df, aes(as.factor(Tooth_Number), Axis1, fill=tclass))  + theme_bw() + ylab("Axis 1") + facet_wrap(~Tooth_Aspect, ncol=1)+ xlab("Tooth") + scale_x_discrete(breaks = seq(from=0, to=35, by=5), labels=seq(from=0, to=35, by=5)) + xlab("Tooth")  + ylim(-8, 8)  + geom_boxplot()+theme(legend.position="none")+ scale_fill_manual(values=cbPalette)+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))

#plot axis 2
a2a = ggplot(xy3.df, aes(as.factor(Tooth_Number), Axis2, fill=tclass))  + theme_bw() + ylab("Axis 2") + facet_wrap(~Tooth_Aspect, ncol=1)+ xlab("Tooth") + scale_x_discrete(breaks = seq(from=0, to=35, by=5), labels=seq(from=0, to=35, by=5)) + xlab("Tooth")  + ylim(-8, 8) +theme(legend.position="none") + geom_boxplot()+ scale_fill_manual(values=cbPalette)+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))


#plot axis 3
a3a = ggplot(xy3.df, aes(as.factor(Tooth_Number), Axis3, fill=tclass))  + theme_bw() + ylab("Axis 3") + facet_wrap(~Tooth_Aspect, ncol=1)+ xlab("Tooth") + scale_x_discrete(breaks = seq(from=0, to=35, by=5), labels=seq(from=0, to=35, by=5)) + xlab("Tooth")  + ylim(-8, 8) +theme(legend.position="none")  + geom_boxplot()+ scale_fill_manual(values=cbPalette)+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))

#plot axis 4
a4a = ggplot(xy3.df, aes(as.factor(Tooth_Number), Axis4, fill=tclass))  + theme_bw() + ylab("Axis 4") + facet_wrap(~Tooth_Aspect, ncol=1)+ xlab("Tooth") + scale_x_discrete(breaks = seq(from=0, to=35, by=5), labels=seq(from=0, to=35, by=5)) + xlab("Tooth")  + ylim(-8, 8) +theme(legend.position="none") + geom_boxplot()+ scale_fill_manual(values=cbPalette)+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))

grid.arrange(a1a, a2a, a3a, a4a, ncol=2)


#how many comparisons are there in the boxplots
#buccal
b = subset(xy3.df, Tooth_Aspect=="Buccal")
table(b$Tooth_Class)
#lingual
l = subset(xy3.df, Tooth_Aspect=="Lingual")
table(l$Tooth_Class)


```


##### 2.4. Generate Figure 2a. 
- Plot axis 1 against axis 2
- Also look at boxplots of tooth class 
```{r}
a1a = ggplot(xy3.df, aes(as.factor(Tooth_Number), Axis1, fill=tclass))  + theme_bw() + ylab("Axis 1") + facet_wrap(~Tooth_Aspect, ncol=1)+ xlab("Tooth") + scale_x_discrete(breaks = seq(from=0, to=35, by=5), labels=seq(from=0, to=35, by=5)) + xlab("Tooth")  + ylim(-8, 8)  + theme(plot.title = element_text(size=12)) + geom_boxplot()+theme(legend.position="none")+scale_fill_manual(values=cbPalette) 
a1a

c1 = ggplot(xy3.df, aes(Axis2, Axis1, colour=Tooth_Class))  + theme_bw()  + geom_point() + guides(color=guide_legend(title="Tooth Class")) + theme(legend.position="none")+ scale_color_manual(values=cbPalette)+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))
c1

ggsave(a1a, file="~/Dropbox/Figure2a.eps",  width = 3.5, height = 6, units ="in",dpi = 300, device="eps")

#ggsave(a1a, file="~/Desktop/Figure2b.png",  width = 3.5, height = 6, units ="in",dpi = 300, device="png")


```


#### 2.5. get the adjusted R^2 value for the complete model and then perform forward-selection using the Blanchet et al (2008) double stopping criterion
- ok so we think we see that there is a trend; now we want to perform forward selection on the polynomial terms
- then identify the spatial terms that explain a significant fraction of the variance in community data
```{r, fig.height=6, fig.width=12}
#third order non-orthogonal polynomial
fullmouth.trendnon.rda <- rda(otus.h ~., data=as.data.frame(poly.xy3))
(R2adj.poly <- RsquareAdj(fullmouth.trendnon.rda)$adj.r.squared)

# do the forward selection
library(packfor)
fullmouth.trend.fwd <- forward.sel(otus.h, poly.xy3, adjR2thresh=R2adj.poly)
kable(fullmouth.trend.fwd) 
#write.csv(fullmouth.trend.fwd, file="~/Dropbox/Figures_20170617/Revised/FinalFigures/TableS5.csv")

#perform a new RDA using the significant terms
fullmouth.trend.rda2 <- rda(otus.h~., data=as.data.frame(poly.xy3)[,fullmouth.trend.fwd[,2]])


#what is the R2 of the new model
(R2adj.fwd <- RsquareAdj(fullmouth.trend.rda2)$adj.r.squared)

set.seed(990)
#test the significance of the model 
fm.aov = anova.cca(fullmouth.trend.rda2, step=1000)
fm.aov

#test the significance of the individual axes
set.seed(88)
fm.aov.axes = anova.cca(fullmouth.trend.rda2, step=1000, by="axis")
fm.aov.axes
#plot the 5 significant spatial structures (i.e., canonical axes)
#note that scaling to 1 displays the spatial model (i.e., the linear combo of spatial variables while preserving the Euclidean distance among sites)
fullmouth.trend.fit <- scores(fullmouth.trend.rda2, choices=c(1, 2, 3, 4), display="lc", scaling=1)
fit = data.frame(fullmouth.trend.fit, sample_data(f1))
fit$x  = as.numeric(as.character(fit$x))
fit$y  = as.numeric(as.character(fit$y))

library(RColorBrewer)
myPalette = colorRampPalette(brewer.pal(11, "RdBu"), space="Lab")

#plot the first axis
p1 = ggplot(fit, aes(x, y, color=RDA1)) +ggtitle("RDA1") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + theme(plot.title = element_text(size=12))

#plot the second axis
p2 = ggplot(fit, aes(x, y, color=RDA2)) +ggtitle("RDA2") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ theme(plot.title = element_text(size=12))


#plot the third axis
p3 = ggplot(fit, aes(x, y, color=RDA3)) +ggtitle("RDA3") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ theme(plot.title = element_text(size=12))

#plot the 4th axis
p4 = ggplot(fit, aes(x, y, color=RDA4)) +ggtitle("RDA4") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ theme(plot.title = element_text(size=12))


#visualize the significant axes fit
grid.arrange(p1, p2, p3,p4, ncol=4)

#ggsave(grid.arrange(p1, p2, p3,p4,ncol=2), file="~/Dropbox/Figures_20170617/Revised/FinalFigures/FigureS7a.png",  width = 10, height = 9, units ="in",dpi = 300, device="png") 
```



#### 2.6. Plot the significant polynomial terms
- by looking at these, we can gain additional insight into the spatial variables that explain variation in community structure
```{r, fig.height=6, fig.width=12}
sig = fullmouth.trend.fwd$variables
sig = str_replace_all(sig, "[(^)]", ".")
xy.sig = data.frame(poly.xy3)[sig]

#subset on the significant polynomial terms
df = as.data.frame(poly.xy3)[,fullmouth.trend.fwd[,2]]
df = data.frame(df, sample_data(f1))
df$x = as.numeric(as.character(df$x))
df$y = as.numeric(as.character(df$y))

fullmouth.trend.fwd$F.Stat = round(fullmouth.trend.fwd$F, 2)


#plot the first axis
t1 = ggplot(df, aes(x, y, color=X.2)) +ggtitle("X^2") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + theme(plot.title = element_text(size=12))

t2 = ggplot(df, aes(x, y, color=Y.2)) +ggtitle("Y^2") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + theme(plot.title = element_text(size=12))

t3 = ggplot(df, aes(x, y, color=Y)) +ggtitle("Y") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + theme(plot.title = element_text(size=12))

t4 = ggplot(df, aes(x, y, color=Y.3)) +ggtitle("Y^3") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + theme(plot.title = element_text(size=12))

t5 = ggplot(df, aes(x, y, color=X.2Y)) +ggtitle("X^2*Y") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + theme(plot.title = element_text(size=12))

t6 = ggplot(df, aes(x, y, color=X.3)) +ggtitle("X^3") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + theme(plot.title = element_text(size=12))

grid.arrange(t1, t2, t3,t4, t5, t6, ncol=3)

#ggsave(grid.arrange(t1, t2, t3,t4, t5, t6, ncol=2), file="~/Dropbox/Figures_20170617/Revised/FinalFigures/FigureS7b.png",  width = 10, height = 9, units ="in",dpi = 300, device="png") 
```


#Figure 2. How do we make sense of the gradient? Let's look at the distribution of taxa across sites

#### 2.8. Figure 2b: Make a between-taxa correlation matrix plot to highlight what the BCA reveals

```{r, fig.width=12, fig.height=12, warning=FALSE}
#merge the temporal replicates within each subject to avoid temporal pseudoreplication        
subs <- levels(sample_data(f1)$Subject) 
holder <-vector('list',length(subs)) 


for(i in 1:length(subs)){ 
    subx = subset_samples(f1, Subject==subs[[i]])
    subh = merge_samples(subx, "Habitat")
    sample_data(subh)$Subject = subs[[i]]
    sample_data(subh)$Habitat = sample_names(subh)
    sample_names(subh) = paste0(sample_data(subh)$Subject, "_", sample_names(subh))
    holder[[i]] = subh
}

is = merge_phyloseq(holder[[1]], holder[[2]], holder[[3]], holder[[4]], holder[[5]], holder[[6]], holder[[7]])
is2 = prune_taxa(taxa_sums(is) > 0, is)
is2 = prune_samples(sample_sums(is2) > 0, is2)
is2

#set up a new mapping file to correct what was lost during the merge
        foo = sample_names(is2)
        foo2 <- str_replace_all(foo, "([B,L])", replacement =c(".Buccal", ".Lingual"))
        foo2 <- str_replace_all(foo2,  "_", ".")
        map <- colsplit(foo2, "([.])", c("Subject", "Junk", "Tooth_Number", "Tooth_Aspect"))
        map = data.frame(sample_data(is2)$Habitat, map)
        colnames(map)[1] = "Habitat"
         
        #merge in the mapping file information
        dot = read.csv("~/Desktop/Proctor_NatureComm/mytoothdot_coordinates_FullMouth.csv")
        colnames(dot)[1] = "Habitat"
        
        #merge maps
        library(plyr)
        map2 = join(map, dot, by="Habitat")
        
        #make the sample names of map2 the same as the names in bs2 then replace the mapping file
        names = paste0(map2$Subject, "_", map2$Habitat)
        rownames(map2) = names
        sample_data(is2) = map2


#subset on the most abundant taxa for ease of visualization
library(ggcorrplot)
sub = names(sort(taxa_sums(is2), TRUE)[1:70]) 
sub2 = prune_taxa(sub, is2)

#stabilize the variance
sub2_VST = sub2
otu_table(sub2_VST) <- otu_table(sub2_VST) + 1
sub2.ds = phyloseq_to_deseq2(sub2_VST, ~Tooth_Class)

#variance stabilizing transformation
sub2_vst <- varianceStabilizingTransformation(sub2.ds , blind=FALSE, fitType="local") 
counts_VST <- otu_table(as.matrix(assay(sub2_vst)), taxa_are_rows=TRUE)
otu_table(sub2_VST) <- counts_VST 


#change the names
tax = data.frame(tax_table(sub2))
names = tax$Highest.Rank
otus= data.frame(otu_table(sub2))

#generate the corrrelation matrix and assign the taxa names to rows and columns
cormat = round(cor(otus), 1)
dim(cormat)
rownames(cormat) = names
colnames(cormat) = names

#generate pvalues
p.mat <- cor_pmat(otus)

p=ggcorrplot(cormat, hc.order = TRUE, type="full", hc.method = "average", lab_size = 6, tl.cex=6, sig.level=0005, colors = c("turquoise1", "white", "coral")) + theme(text = element_text(size=12), axis.text.x = element_text(angle=90, vjust=1)) +theme(legend.position="bottom")


#test figure
p
ggsave(p, file="~/Dropbox/Figure2b.eps",  width = 10, height = 8, units ="in",dpi = 300, device="eps")
```


### Figure 2c: plot the distribution of select taxa that vary significantly across sites 
```{r, fig.width=6, fig.height=6, warning=FALSE}
keep = c("Corynebacterium durum", "Rothia dentocariosa", "Veillonella dispar str. 01", "Haemophilus parainfluenzae",  "Rothia aeria")


#rename taxa names with highest rank
tax = data.frame(tax_table(is2))
temp = is2
taxa_names(temp) = tax$Highest.Rank
sub1 = prune_taxa(keep, temp)

CS1.phys = sub1
  #get the data into data.frame
  CS1.df <- data.frame(tax_table(CS1.phys), t(otu_table(CS1.phys)),
                       taxa_names(CS1.phys))
  colnames(CS1.df)[395] <- "RSV"
  CS1.dfm <- melt(CS1.df, id.var=c("RSV", colnames(tax_table(CS1.phys))))
  
  #make a denovo mapping file to fix what was lost during merge
  map1 = read.csv("~/Desktop/Proctor_NatureComm/mytoothdot_coordinates_FullMouth.csv")

  colnames(map1)[1] <- "Habitat"
  
  fm = CS1.dfm$variable
  fm1 = str_replace_all(fm, "_S", ".S")
  fm2 = colsplit(fm1, "[(.)]", c("J1", "J2", "Habitat"))
  CS1.dfm$Habitat = fm2$Habitat
  
  CS1.dfm = join(CS1.dfm, map1, by="Habitat")
  CS1.dfm$Tooth_Number <- as.factor(CS1.dfm$Tooth_Number)
#re-create the subject variable
foo = CS1.dfm$variable 
foo2 = colsplit(foo, "[(_)]", c("Subject", "Site"))
foo2 = str_replace_all(foo2$Subject, "X", "")
CS1.dfm$Subject = foo2

CS1.dfm$value = CS1.dfm$value + 1

f=ggplot(CS1.dfm, aes(as.numeric(Tooth_Number), value, group=Highest.Rank, color=Highest.Rank))  + geom_smooth(se=TRUE) + theme_bw() + facet_wrap(~Tooth_Aspect, ncol=1)+ scale_y_continuous(trans='log2') + xlab("Tooth Number") + ylab("Log2 Abundance")  + guides(color=guide_legend(title="Taxon"))   + scale_x_continuous(breaks = seq(from=0, to=30, by=5), labels=seq(from=0, to=30, by=5))+theme(legend.position="none")
f
ggsave(f, file="~/Dropbox/Figure2c.png",  width = 3.5, height = 6, units ="in",dpi = 300, device="png")  
#note: i printed the p and f plots and put together the figure in powerpoint
```



#### 4. Figure 3: Do communities vary across a gradient on the oral mucosa also (i.e., mucosal biogeography dataset)?
The ordination method presented so far demonstrate that the data conform to a spatial pattern, but none of these models explicitly model the variance as a function of geographic location. Moreover, we also  wanted to know whether communities inhabiting the mucosal surfaces (e.g., buccal mucosa, alveolar mucosa, keratinized gingiva) also vary along an ecological gradient.

To examine the variation of OTUs across a spatial gradient, we performed another PCA-IV, but instead of constraining by a factor (e.g., Habitat), we constrained by a third order polynomial function of the geographic coordinates. 
- Constrain ordination of the PCA Correlation Matrix by the polynomial function
- Polynomial function: Y ~ X + X^2 + X^3 + Y + XY + X^2Y + Y^2 + XY^2 + Y^3
- Maximize the variance explained by the polynomial terms 
- Sometimes called Trend Surface Analysis
- We see that the first 4 axes explain a chunk of the variation 

```{r, Fig 3, fig.width=12, fig.height=6, warning=FALSE}
#convert the biogeography datatset to trelative abundance
bioh.otus = data.frame(otu_table(Biogeo2_phys15))
bioh.euc = vegdist(bioh.otus, "euclidean")
bioh.nmds = cmdscale(bioh.euc, eig=TRUE, k=10)
bioh.map = data.frame(sample_data(Biogeo2_phys15), scores(bioh.nmds))
h.map$method = "Hellinger"
h.eig = 100*(h.nmds$eig/sum(h.nmds$eig))
evals = h.nmds$eig
head(h.eig)

df = data.frame(h.eig[1:10], 1:10)
colnames(df) = c("Percent.Variation", "rank")

p=ggplot(df, aes(rank, Percent.Variation)) + geom_point() + theme_bw()

p
```


####3. Figure 3, Do PCA-IV on geographic coordinates to see if gradient is evident in mucosal communities
#get environmental data
```{r}
#coordinates
map <- sample_data(Biogeo2_VST)
coo <- cbind(map$x, map$y)
colnames(coo) <- c("x", "y")

#hellinger transform the data
otus = otu_table(Biogeo2_phys15)
otus.h = decostand(otus, "hellinger")

### Perform  PCA-IV with respect to a 3rd order polynomial of geographic coordinates
poly.xy3 <- poly(coo, degree = 3, raw=FALSE) #, coefs=TRUE) 
colnames(poly.xy3) <- c("X", "X^2", "X^3", "Y", "XY", "X^2Y", "Y^2", "XY^2", "Y^3")
poly.xy3.df <- data.frame(poly.xy3, map$x, map$y)

library(ade4)
#totus <- data.frame(t(otu_table(Biogeo2_VST)))
rld.pca <- dudi.pca(otus.h, center=TRUE, scale=TRUE, scannf=FALSE, nf=10)
rld.xy3 <- pcaiv(rld.pca, poly.xy3, scannf = FALSE, nf = 6)
rld.xy3.df <- data.frame(rld.xy3$ls, map)
xy3.df <- data.frame(rld.xy3.df, poly.xy3)

# how much of the variance does this model explain?
rld.xy3.var <- sum(rld.xy3$eig)/sum(rld.pca$eig)*100
rld.xy3.var
# get the eigen values
pca.scree <- data.frame(rank = 1:length(rld.xy3$eig))
Explainedvariance = rld.xy3$eig/sum(rld.xy3$eig)*100
Explainedvariance
#how much variance does the first axis explain?
(Explainedvariance[1]/100)*rld.xy3.var
#force x, y to numeric
xy3.df$x <- as.numeric(as.character(xy3.df$x))
xy3.df$y <- as.numeric(as.character(xy3.df$y))

### Force variables to order in ggplot
xy3.df$Tooth<- as.numeric(as.character(xy3.df$Tooth))


#order the sites for plotting
order <- c("Alveolar Mucosa   ", "Keratinized Gingiva   ", "Supragingival Tooth   ", "Buccal Mucosa   ")
xy3.df$Site.New <- factor(xy3.df$Site.New, as.character(order))
xy3.df$Class = str_replace_all(xy3.df$Class, "CentralIncisor", "Incisor")
xy3.df$Class = str_replace_all(xy3.df$Class, "LateralIncisor", "Incisor")
cbPalette <- c("grey76", "Salmon", "#00BFC4", "grey76")


#plot Axis 1 over teeth 
fig3 = ggplot(xy3.df, aes(Tooth, Axis1)) +geom_smooth(fill = "gray90", alpha=1) + geom_point(aes(colour=Class)) + scale_colour_manual(values=cbPalette) +facet_wrap(Aspect~Site.New, ncol=4, scales="free_y") + theme_bw() + ylab("Axis 1") + xlab("Tooth") + ggtitle("")+ theme(plot.title = element_text(size=15)) +scale_x_continuous(breaks = seq(from=0, to=35, by=5), labels=seq(from=0, to=35, by=5))


ggsave(fig3, file="~/Dropbox/Figure3.eps",  width = 10, height = 8, units ="in",dpi = 300, device="eps")  

print(fig3)

```

#### 3.1. Do forward selection of the polynomial terms for the RDA
```{r, fig.height=6, fig.width=12}
#third order non-orthogonal polynomial
fullmouth.trendnon.rda <- rda(otus.h ~., data=as.data.frame(poly.xy3))
(R2adj.poly <- RsquareAdj(fullmouth.trendnon.rda)$adj.r.squared)

# do the forward selection
library(packfor)
fullmouth.trend.fwd <- forward.sel(otus.h, poly.xy3, adjR2thresh=R2adj.poly)
fullmouth.trend.fwd 

#perform a new RDA using the significant terms
fullmouth.trend.rda2 <- rda(otus.h~., data=as.data.frame(poly.xy3)[,fullmouth.trend.fwd[,2]])


#what is the R2 of the new model
(R2adj.fwd <- RsquareAdj(fullmouth.trend.rda2)$adj.r.squared)


#test the significance of the model 
set.seed(44)
bio.aov = anova.cca(fullmouth.trend.rda2, step=1000)
bio.aov

#test the significance of the individual axes
set.seed(99)
axes.aov = anova.cca(fullmouth.trend.rda2, step=1000, by="axis")
axes.aov

#plot the 5 significant spatial structures (i.e., canonical axes)
#note that scaling to 1 displays the spatial model (i.e., the linear combo of spatial variables while preserving the Euclidean distance among sites)
fullmouth.trend.fit <- scores(fullmouth.trend.rda2, choices=c(1, 2, 3, 4, 5), display="lc", scaling=1)
fit = data.frame(fullmouth.trend.fit, sample_data(Biogeo2_phys15))
fit$x  = as.numeric(as.character(fit$x))
fit$y  = as.numeric(as.character(fit$y))

library(RColorBrewer)
myPalette = colorRampPalette(brewer.pal(11, "RdBu"), space="Lab")

#plot the first axis
p1 = ggplot(fit, aes(x, y, color=RDA1)) +ggtitle("RDA1") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw() + facet_wrap(~Site.New, ncol=4)

#plot the second axis
p2 = ggplot(fit, aes(x, y, color=RDA2)) +ggtitle("RDA2") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)


#plot the third axis
p3 = ggplot(fit, aes(x, y, color=RDA3)) +ggtitle("RDA3") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)



grid.arrange(p1, p2, p3, ncol=3)
```


##### 3.2. Supplemental Figure S11: Plot the other significant axes for the mucosal biogeography dataset
```{r, fig.width=12, fig.height=8}
xy3.df$Site.New = str_replace_all(xy3.df$Site.New, "Alveolar Mucosa   ", "AM")
xy3.df$Site.New = str_replace_all(xy3.df$Site.New, "Keratinized Gingiva   ", "KG")
xy3.df$Site.New = str_replace_all(xy3.df$Site.New, "Supragingival Tooth   ", "Tooth")
xy3.df$Site.New = str_replace_all(xy3.df$Site.New, "Buccal Mucosa   ", "BM")
  
a2 =  ggplot(xy3.df, aes(Tooth, Axis2))  + theme_bw() + ylab("Axis 2") + facet_wrap(Aspect~Site.New, ncol=4, scales="free_y")+ xlab("Tooth") + xlab("Tooth")   +theme(legend.position="none") + theme(plot.title = element_text(size=8)) + geom_smooth() + geom_point()   + theme(text = element_text(size=10), axis.text.x = element_text(angle=00, vjust=1))

a3 =  ggplot(xy3.df, aes(Tooth, Axis3))  + theme_bw() + ylab("Axis 3") + facet_wrap(Aspect~Site.New, ncol=4, scales="free_y")+ xlab("Tooth") + xlab("Tooth")   +theme(legend.position="none") + theme(plot.title = element_text(size=8)) + geom_smooth() + geom_point()+ theme(text = element_text(size=10), axis.text.x = element_text(angle=00, vjust=1))

grid.arrange(a2, a3, ncol=2)
#ggsave(grid.arrange(a2, a3, ncol=1), file="~/Dropbox/Figures_20170617/Revised/FinalFigures/FigureS11a.png",  width = 6, height = 9, units ="in",dpi = 300, device="png")
```




#### 3.3. Plot the significant polynomial terms
- by looking at these, we can gain additional insight into the spatial variables that explain variation in community structure
```{r, fig.height=6, fig.width=12}
sig = fullmouth.trend.fwd$variables
sig = str_replace_all(sig, "[(^)]", ".")
xy.sig = data.frame(poly.xy3)[sig]

#subset on the significant polynomial terms
df = as.data.frame(poly.xy3)[,fullmouth.trend.fwd[,2]]
df = data.frame(df, sample_data(Biogeo2_phys15))
df$x = as.numeric(as.character(df$x))
df$y = as.numeric(as.character(df$y))
dfm = melt(df, id.vars = colnames(sample_data(Biogeo1_phys)))
#plot the first axis
p1 = ggplot(df, aes(x, y, color=X.2)) +ggtitle("X^2") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)+theme(legend.position = "none")

p2 = ggplot(df, aes(x, y, color=Y.2)) +ggtitle("Y^2") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)+theme(legend.position = "none")

p3 = ggplot(df, aes(x, y, color=Y)) +ggtitle("Y") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)+theme(legend.position = "none")

p4 = ggplot(df, aes(x, y, color=Y.3)) +ggtitle("Y^3") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)+theme(legend.position = "none")

p5 = ggplot(df, aes(x, y, color=X.2Y)) +ggtitle("X^2*Y") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)+theme(legend.position = "none")

p6 = ggplot(df, aes(x, y, color=X)) +ggtitle("X") + geom_point(aes(x=x, y=y)) + coord_fixed() +scale_x_continuous(limits=c(-1,1)) + geom_jitter(position=position_jitter(width=0.025, height = 0.025), size=3) + scale_color_gradientn(colours=myPalette(100)) + theme_bw()+ facet_wrap(~Site.New, ncol=4)+theme(legend.position = "none")

grid.arrange(p1, p2, p3,p4, p5, p6, ncol=2)
#ggsave(grid.arrange(p1, p2, p3,p4, p5, p6, ncol=2),file="~/Dropbox/Figures_20170617/Revised/FinalFigures/FigureS11b.png",  width = 10, height = 9, units ="in",dpi = 300, device="png")
```

#### 5.1. Prelude to Figure 5,  Does low salivary flow impact  the spatial organization of taxa? 
First, let's take a look at the differences between the salivary flow rates of the control and SS populations. We'll look at the Unstimulated Whole Saliva Flow Rate (UWS-FR) as well as the stimulated whole salivary flow rate (SWS-FR).

- Import the data from redcap then merge with the study-specific identifiers
- Add an aim variable to the dataset
- To avoid treating repeated measures as independent samples for the aim 1 controls, we'll get the mean of the flow rates for each subject and then perform Welch's two-sample t-test comparing flow rates between groups. 
- This analysis shows that both the UWS-FR and and the SWS-FR varies significantly between the two groups
- We see that the UWS-FR within each subject is highly corrrelated with the SWS-FR within that same subject
- We also note that one SS subject has a flow rate that's comparable to controls
- Note redcap 63 didn't have an UWS-FR taken on day 1 because of a procedural deviation
- Redcap 239 (aim 3) has a negative UWS-FR for d01; it is based on one measurement only
- Redcap 248 (aim 3) is assumed to have a collection period of 5 minutes
- Redcap 219 is missing d01 flow rates; UWS-FR is based on one measurement only
- Redcap 303 SWS-FR is missing and so the value could not be determined


```{r setup, fig.width=12, fig.height=6, warning=FALSE}
#read in the pids
data1 = read.csv("~/Desktop/Proctor_NatureComm/pids.csv")
redcap.id = levels(as.factor(data1$redcap_record_num))

#read in the flow rates
data2 = read.csv("/Users/dmap/Desktop/Proctor_NatureComm/flow_rates_20170710_v2.0.csv")

#merge the pids with the flow rates
data3 = join(data1, data2, "redcap_record_num")

#add aim variable
sa1 = c(17 , 18, 36 , 39 , 49 , 63 , 170)
data3$aim = ifelse(data3$redcap_record_num %in% sa1, "Control", "Sjogrens")
data3$dcoll_sws_flowrate = as.numeric(as.character(data3$dcoll_sws_flowrate))
data3$dcoll_uws_flowrate = as.numeric(as.character(data3$dcoll_uws_flowrate))

#look at it
head(data3)

#how correlated is UWS-FR with SWS-FR?
p2 =ggplot(data3, aes(dcoll_uws_flowrate, dcoll_sws_flowrate, color=aim)) + geom_point() + theme_bw() + ggtitle("Correlation UWS-FR: SWS-FR")
p2

cor.test(data3$dcoll_uws_flowrate, data3$dcoll_sws_flowrate, method="pearson", exact=TRUE,na.rm=TRUE)


#now let's look at some boxplot
dfm = melt(data3, id.var=c("redcap_record_num", "pid",  "redcap_event_name",  "Notes", "Day", "aim"))
dfm = subset(dfm, value !="NA")
dfm = subset(dfm, value !="ND")
dfm$value = as.numeric(as.character(dfm$value))

head(dfm)



#get the mean flow rates
library(doBy)

#make an uws-fr df
uws = subset(dfm, variable %in% c("dcoll_uws_flowrate", "escr_uws_fr"))
#make a sws-fr df
sws = subset(dfm, variable == "dcoll_sws_flowrate")

#get significance of between group means - UWS
a1_uws = subset(uws, aim=="Control")
a3_uws = subset(uws, aim=="Sjogrens")
t.test(a1_uws$value,a3_uws$value)

#get significance of between group means - UWS
a1_sws = subset(sws, aim=="Control")
a3_sws = subset(sws, aim=="Sjogrens")
t.test(a1_sws$value,a3_sws$value)

#generate the average subject value for the flow rates in aim 1 since we have repeated measures and don't want those to be treated as if they were independent in the t.test
uws_df = summaryBy(value~pid+redcap_record_num, data=uws, FUN=mean, na.rm=TRUE)
sws_df = summaryBy(value~pid+redcap_record_num, data=sws, FUN=mean, na.rm=TRUE)

#merge the mean uws-fr, sws-fr, pid and microbial data
frs = join (uws_df, sws_df, by=c("pid", "redcap_record_num"))

colnames(frs)[1] = c("Subject")
colnames(frs)[3:4] = c("UWS_FR", "SWS_FR")

### Merge the flow rate data with the microbial data
flow <- subset_samples(supra, Day=="1" & Protocol=="Clinic")
flow # 480 taxa and 923 samples
flow.map <- data.frame(sample_data(flow))


#merge the sequencing data with the clinical data
flow.map2 = join(flow.map, frs, "Subject")
```

#### 5.2. Bring in the clinical data - in this section, we bring in caries 
- note that we examined the occlusal surfaces as well, but those data are not represented here since we only sampled the buccal and lingual surfaces of teeth
- We see that in general the caries experience of SS subjects is widespread whereas for the control subjects it's limited to the molars and incisors
- Important to note that these data only represent the smooth surfaces!!

```{r, fig.width=12, fig.height=6, warning=FALSE}
#read in the caries data and then subset on the redcap ids that are included in the dataset
data = read.csv("~/Desktop/Proctor_NatureComm/data_caries_combined_v1.2_20170328.csv")
colnames(data)[3] = "redcap_record_num"
data1 = subset(data, redcap_record_num %in% redcap.id)

#subset on buccal and lingual (excluding the occlusal surfaces as well as the mesial- and distal- sites) and get rid of wisdom teeth
bl = subset(data1, surface %in% c("B", "L"))
nowise = c(2:15, 18:31)
bl = subset(bl, tooth %in% nowise)

#convert disease to 1, sound to 0
bl$binary = ifelse(bl$description =="SOUND", 0, 1)
bl$binary = as.factor(as.numeric(bl$binary))

#count the number of decayed missing teeth per subject
bl.sum = summaryBy(binary~redcap_record_num, data=bl, FUN=sum)
bl.sum$prop = bl.sum/28
colnames(bl)[3]= "redcap_record_num"
colnames(bl)[8]= "Tooth_Number"
colnames(bl)[9]= "Tooth_Aspect"
bl$Tooth_Aspect = str_replace_all(bl$Tooth_Aspect, "([B,L])", replacement =c("Buccal", "Lingual"))
bl = bl[,3:11]


#join the caries data with the microbial data
total3 = join(flow.map2, bl, b=c("redcap_record_num", "Tooth_Number", "Tooth_Aspect"))
total3$x <- as.numeric(as.character(total3$x))
total3$y <- as.numeric(as.character(total3$y))
rownames(total3) = total3[,1]
total3 = sample_data(total3)

# relabel the subjects so they will correspond to the way they are labeled in figure 5
#relabel controls that have symetrical gradients
total3$Subject = str_replace_all(total3$Subject, "1-101", "Control 01")
total3$Subject = str_replace_all(total3$Subject, "1-103", "Control 02") 
total3$Subject = str_replace_all(total3$Subject, "1-104", "Control 03") 
total3$Subject = str_replace_all(total3$Subject, "1-106", "Control 04") 
total3$Subject = str_replace_all(total3$Subject, "1-107", "Control 05") 

#relabel the controls that have flat maxillary lines
total3$Subject = str_replace_all(total3$Subject, "1-102", "Control 06")
total3$Subject = str_replace_all(total3$Subject, "1-105", "Control 07") 


#relabels sjogrens that have symetrical curves
total3$Subject = str_replace_all(total3$Subject, "3-305", "Sjogrens 01") 
total3$Subject = str_replace_all(total3$Subject, "3-302", "Sjogrens 02") 
total3$Subject = str_replace_all(total3$Subject, "3-306", "Sjogrens 03") 


#relabel sjogren's that have flat lines
total3$Subject = str_replace_all(total3$Subject, "3-304", "Sjogrens 04") 
total3$Subject = str_replace_all(total3$Subject, "3-308", "Sjogrens 05") 
total3$Subject = str_replace_all(total3$Subject, "3-310", "Sjogrens 06") 
total3$Subject = str_replace_all(total3$Subject, "3-309", "Sjogrens 07") 


total3$Subject = str_replace_all(total3$Subject, "3-303", "Sjogrens 08") 
total3$Subject = str_replace_all(total3$Subject, "3-307", "Sjogrens 09") 

#relabel sjogrens that have exaggerated  lines

total3$Subject = str_replace_all(total3$Subject, "3-301", "Sjogrens 10") 


#make the total3 variable the new mapping file for the object flow
sample_data(flow) = total3
```


#### 5.3. Figure 5,  Does salivary flow impact the structure of bacterial communities?
- all flow rates but the one for redcap 39 are from the ESE
- flow rate for redcap 39 is from day 1 of sample collection since data were missing for the ESE
```{r, fig.width=12, fig.height=6, warning=FALSE}
#how many samples and subjects are there?
flow 
levels(sample_data(flow)$Subject) #7 controls; 10 aim 3

#get ridof subjects with missing flow rate for the cca
flows = subset_samples(flow, UWS_FR & SWS_FR != "NA")
levels(sample_data(flows)$Subject) #7 controls; lost subject 3-310 because of sws-fr


#subset on taxa present in 10% of samples
filtergroup = filterfun(kOverA(k=75, A=1)) #k = number of samples; A = abundance
        flows = filter_taxa(flows, filtergroup, prune=TRUE) 
        flows = prune_taxa(taxa_sums(flows) > 0, flows)
        flows = prune_samples(sample_sums(flows) > 0, flows)
        flows
        
#stabilize the variance
flows_VST = flows
otu_table(flows_VST) <- otu_table(flows_VST) + 1
flows.ds = phyloseq_to_deseq2(flows_VST, ~PoolName)

#variance stabilizing transformation
flows_vst <- varianceStabilizingTransformation(flows.ds , blind=FALSE, fitType="local") 
counts_VST <- otu_table(as.matrix(assay(flows_vst)), taxa_are_rows=TRUE)
otu_table(flows_VST) <- counts_VST 


#set an analysis up for CCA
otus = t(otu_table(flows_VST))
map1 = data.frame(sample_data(flows_VST)$Tooth_Class, sample_data(flows_VST)$UWS_FR, sample_data(flows_VST)$SWS_FR, sample_data(flows_VST)$binary)
colnames(map1) = c("Class", "uws_fr", "sws_fr", "dmfs")
map1$Class = as.numeric(map1$Class)
subs = as.vector(sample_data(flows_VST)$Subject)
runs = as.vector(sample_data(flows_VST)$Pool_Name)

#for some reason this wworks on the command line but not in knitr
attach(map1)
flow.cca = vegan::cca(otus ~ map1$Class*map1$uws_fr*map1$sws_fr*map1$dmfs, sitenv=map1)

#run an anova on this
set.seed(100)
flow.aov = anova.cca(flow.cca, by="term", strata=subs)
flow.aov

#how much variance does the model explain
eig = flow.cca$CCA$eig
eigp = 100*(eig/sum(eig))
head(eigp)

#how much does the first axis explain?
eigp[1]

#hwat about the second?
eigp[2]


#get the sample site scores and plot them
sites = data.frame(flow.cca$CCA$wa, sample_data(flows_VST))
sites$Aim = ifelse(sites$Aim=="SA1", "Control", "Sjogrens")


#convert the MFS score to factor
sites$binary = as.factor(as.numeric(sites$binary))

p1 = ggplot(sites, aes(CCA1, CCA2, color=as.factor(Aim))) + geom_point() + theme_bw() + guides(color=guide_legend(title="Health Status")) + ggtitle("a)")  + xlab("CCA1 (59.74%)") + ylab("CCA2 (15.29%)") + theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))

p2 = ggplot(sites, aes(CCA1, CCA2, color=uws_fr)) + theme_bw() + guides(color=guide_legend(title="UWS-FR")) + ggtitle("b)") + scale_colour_gradientn(colours=c("red", "blue")) + labs(title = "b)" ) + xlab("CCA1 (59.74%)") + ylab("CCA2 (15.29%)")+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1)) + geom_point()

p3 = ggplot(sites, aes(CCA1, CCA2, color=sws_fr))  + theme_bw() + guides(color=guide_legend(title="SWS-FR")) + ggtitle("c)") + scale_colour_gradientn(colours=c("red", "blue")) + labs(title = "c)" )+ xlab("CCA1 (59.74%)") + ylab("CCA2 (15.29%)")+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))+ geom_point()

p4 = ggplot(sites, aes(CCA1, CCA2, color=as.factor(binary))) + theme_bw() + guides(color=guide_legend(title="MFS")) + ggtitle("d)")  + labs(title = "d)" )+ xlab("CCA1 (59.74%)") + ylab("CCA2 (15.29%)")+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))+ geom_point()+ scale_colour_manual(values = c("red", "blue"))

p5 = ggplot(sites, aes(uws_fr, CCA1, color=Aim)) + geom_jitter() +theme_bw() + ggtitle("e)") + ylab("CCA1 (59.74%)") +xlab("UWS-FR")+ guides(color=guide_legend(title="Health Status"))+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))

p6 = ggplot(sites, aes(sws_fr, CCA1, color=Aim)) + geom_jitter() +theme_bw() + ggtitle("f)")  +xlab("SWS-FR") + ylab("CCA1 (59.74%)") + guides(color=guide_legend(title="Health Status"))+ theme(text = element_text(size=10, family="Helvetica"), axis.text.x = element_text(angle=00, vjust=1))


fig5= arrangeGrob(p1, p2,  p3, p4, p5, p6, ncol=3)
ggsave(grid.arrange(p1, p2, p3, p4, p5, p6, ncol=3), file="~/Dropbox/Figure4.eps",  width = 11, height = 8.5, units ="in",dpi = 300)

grid.arrange(p1, p2, p3, p4, p5, p6, ncol=3)
```


##5.4. look at the distribution of SS samples along axis 1
```{r}
ss = subset(sites, Aim=="Sjogrens")
### how many ss samples map to negative axis 1 scores
neg = subset(ss, CCA1 < 0)
ss.neg= subset(neg, Aim=="Sjogrens")
nrow(ss.neg)/nrow(ss)
### how many ss samples map to positive axis 1 scores
pos = subset(ss, CCA1 > 0)
ss.pos= subset(pos, Aim="Sjogrens")
nrow(ss.pos)/nrow(ss)

########## look at the distribution of control samples along axis 1
cont = subset(sites, Aim=="Control")
### how many ss samples map to negative axis 1 scores
neg = subset(cont, CCA1 < 0)
nrow(neg)/nrow(cont)

### how many ss samples map to positive axis 1 scores
pos = subset(cont, CCA1 > 0)
nrow(pos)/nrow(cont)

```


#### 6. Figure 6,  Does low salivary flow impact  the spatial organization of taxa? 

```{r, fig.height=12, fig.width=8}
#how many samples and subjects are there?
flow 
levels(sample_data(flow)$Subject) #7 controls; 10 aim 3

#stabilize the variance
flows_VST = flow
otu_table(flows_VST) <- otu_table(flows_VST) + 1
flows.ds = phyloseq_to_deseq2(flows_VST, ~PoolName)

#variance stabilizing transformation
flows_vst <- varianceStabilizingTransformation(flows.ds , blind=FALSE, fitType="local") 
counts_VST <- otu_table(as.matrix(assay(flows_vst)), taxa_are_rows=TRUE)
otu_table(flows_VST) <- counts_VST 

#transform sample counts for the PCA-IV
library(ade4)
otus = data.frame(otu_table(flow))
otus.h = decostand(otus, "hellinger")
map <- sample_data(flow)
        coo <- cbind(map$x, map$y)
        colnames(coo) <- c("x", "y")
 
### Perform  PCA-IV with respect to a 3rd order polynomial of geographic coordinates
poly.xy3 <- poly(coo, degree = 3, raw=FALSE) #, coefs=TRUE) 
colnames(poly.xy3) <- c("X", "X^2", "X^3", "Y", "XY", "X^2Y", "Y^2", "XY^2", "Y^3")
        poly.xy3.df <- data.frame(poly.xy3, map$x, map$y)
        library(ade4)
        #totus <- t(otu_table(flows_VST))
        rld.pca <- dudi.pca(otus.h, center=TRUE, scale=TRUE, scannf=FALSE, nf=5)
        rld.xy3 <- pcaiv(rld.pca, poly.xy3, scannf = FALSE, nf = 2)
        rld.xy3.df <- data.frame(rld.xy3$ls, map)
        xy3.df <- data.frame(rld.xy3.df, poly.xy3)

# how much of the variance does this model explain?
rld.xy3.var <- sum(rld.xy3$eig)/sum(rld.pca$eig)*100
        rld.xy3.var

# get the eigen values
pca.scree <- data.frame(rank = 1:length(rld.xy3$eig))
        Explainedvariance = rld.xy3$eig/sum(rld.xy3$eig)*100
        Explainedvariance

#organize the data for plotting
xy3.df$x <- as.numeric(as.character(xy3.df$x))
        xy3.df$y <- as.numeric(as.character(xy3.df$y))
        xy3.df$Tooth_Number <- as.numeric(as.character(xy3.df$Tooth_Number))
        #force the tooth to order
        ordering = 1:32
        xy3.df$interval <- factor(xy3.df$Tooth_Number, as.character(ordering))
        xy3.df$Jaw = ifelse(xy3.df$Jaw_Quadrant %in% c(1, 2), "Maxilla", "Mandible")
        xy3.df$Aim = ifelse(xy3.df$Aim=="SA1", "Control", "Sjogrens")
        upper <- 1:2
        lower <- 3:4
        upper.df <- subset(xy3.df, Jaw_Quadrant %in% upper)
        lower.df <- subset(xy3.df, Jaw_Quadrant %in% lower)

#replace subject labels for plotting
xy3.df$Subject = as.factor(xy3.df$Subject)


colnames(xy3.df)[41] = "MFS"
xy3.df$MFS = as.factor(as.numeric(xy3.df$MFS))

xy3.df$Tooth_Class = str_replace_all(xy3.df$Tooth_Class, "Molar_Pre", "Pre-molar")
xy3.df$Tooth_Class = str_replace_all(xy3.df$Tooth_Class, "Incisor_Central", "Central Incisor")
xy3.df$Tooth_Class = str_replace_all(xy3.df$Tooth_Class, "Incisor_Lateral", "Lateral Incisor")

#generate figure
fig5 = ggplot(xy3.df, aes(Tooth_Number, Axis1, color=UWS_FR)) +  geom_smooth(fill = "gray90", alpha=1) + facet_wrap(~Subject, ncol=6, scales="free") + theme_bw() + ylab("Axis 1")   + guides(fill=guide_legend(title="Health Status")) + xlab("Tooth Number") + geom_point(aes(shape=MFS), size=2) + ggtitle("")+ scale_colour_gradientn(colours=c("red", "blue")) 

fig5
ggsave(fig5, file="~/Dropbox/Figure5.eps",  width = 11, height = 8.5, units ="in",dpi = 300, device="eps")  
#save.image(file="Proctor_MainFigures_v28.0.Rdata")
```

### Do molars and incisors differentiate for subjects with sjogren's compared to controls?
- do a wilcoxon rank sum test on axis 1 scores to see if sjogren's samples differentiate along axis 1, as do control samples
```{r}
library("ggpubr")
df  = fig5$data
df= subset(df, Tooth_Class %in% c("Central Incisor", "Molar"))
buccal = subset(df, Tooth_Aspect=="Buccal")

p=ggplot(df, aes(Tooth_Class, Axis1, color=Tooth_Class)) + geom_boxplot() + facet_wrap(~Subject) + theme_bw()+ theme(text = element_text(size=12), axis.text.x = element_text(angle=90, vjust=1))
p + stat_compare_means(method = "wilcox.test")
```
